dsp_permissions_scripts/doap/doap_model.py

```diff
@@ -38,6 +38,27 @@ def assert_correct_combination(self) -> Self:
         return self
 
 
+class NewDoapTarget(BaseModel):
+    """Represents the target of a DOAP that is yet to be created."""
+
+    group: Group | None = None
+    resource_class: str | None = None
+    property: str | None = None
+
+    @model_validator(mode="after")
+    def assert_correct_combination(self) -> Self:
+        # asserts that DOAP is only defined for Group or ResourceClass or Property
+        # or a combination of ResourceClass and Property
+        match (self.group, self.resource_class, self.property):
+            case (None, None, None):
+                raise ValueError("At least one of group, resource_class or property must be set")
+            case (_, None, None) | (None, _, _):
+                pass
+            case _:
+                raise ValueError("Invalid combination of group, resource_class and property")
+        return self
```

Comment:
non-blocking: looking at this model, the logic is "either group is defined, in which case res and prop must be None; or some combination of res and prop is defined, in which case group must be none".  
The cleaner way to represent this logic in a model, is to separate these two core concepts, i.e. to say a `NewDoapTarget` is either a `GroupDoapTarget` (which is defined by the group) or an `EntityDoapTarget` (which is defined by a combination of res and prop, but a restriction that these are both optional but not both can be None).  
The model defined here is perfectly sufficient, this is just a theoretical remark about "how to model data cleanly)

Reply:
to be honest, I'm not sure it will be pleasing, especially since Python is not too good as expressing enum types, so this may result in excessive nesting of some sort. But I'm happy to see it tried out, let me know how it goes